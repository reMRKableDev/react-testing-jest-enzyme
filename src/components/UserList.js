import React from "react";
import User from "./User";

const UserList = ({ retrievedUsers }) => {
  return retrievedUsers.map(user => <User key={user.id} user={user} />);
};

export default UserList;
