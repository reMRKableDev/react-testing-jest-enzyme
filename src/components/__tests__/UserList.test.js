import React from "react";
import { shallow } from "enzyme";
import UserList from "../UserList";

test("shallow rendering of component", () => {
  const fakeUserList = [
    { id: 255, name: "Malcolm Kente", email: "malcolm.kente@remrkabledev.com" },
    { id: 80, name: "Kendrick Lamar", email: "k.dot@compton.com" },
    { id: 444, name: "Shawn Carter", email: "hova@rocnation.com" }
  ];
  const wrapper = shallow(<UserList retrievedUsers={fakeUserList} />);
  expect(wrapper).toMatchSnapshot();
});
