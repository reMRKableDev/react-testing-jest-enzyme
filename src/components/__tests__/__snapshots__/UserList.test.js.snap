// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow rendering of component 1`] = `
Array [
  <User
    key="255"
    user={
      Object {
        "email": "malcolm.kente@remrkabledev.com",
        "id": 255,
        "name": "Malcolm Kente",
      }
    }
  />,
  <User
    key="80"
    user={
      Object {
        "email": "k.dot@compton.com",
        "id": 80,
        "name": "Kendrick Lamar",
      }
    }
  />,
  <User
    key="444"
    user={
      Object {
        "email": "hova@rocnation.com",
        "id": 444,
        "name": "Shawn Carter",
      }
    }
  />,
]
`;
